queries/analysis/class-partition.rq

```diff
@@ -2,17 +2,17 @@ PREFIX void: <http://rdfs.org/ns/void#>
 
 CONSTRUCT {
     ?dataset a void:Dataset ;
-        void:classPartition [
-            void:class ?type ;
-            void:entities ?entities ;
-        ] .
+        void:classPartition ?classPartition .
+    ?classPartition void:class ?type ;
+        void:entities ?entities .
 }
 #namedGraph#
 WHERE {
-    SELECT (COUNT(?type) AS ?entities) ?type {
+    SELECT (COUNT(?type) AS ?entities) ?type ?classPartition {
         #subjectFilter#
         ?s a ?type .
+        BIND(URI(CONCAT("#class-", MD5(STR(?type)))) as ?classPartition)
```

Comment:
Does using a hash make sense? Of course, MD5 is prone to collisions but that shouldn’t matter in this local scope. Alternatively, use `ENCODE_FOR_URI`, but that’s less compact in most cases.